packages/client/src/pages/home/map.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect } from "react";
+import {
+  GoogleMap,
+  LoadScript,
+  Marker,
+  Autocomplete,
+} from "@react-google-maps/api";
+import { CSSProperties } from "@material-ui/styles";
+import { useAppDispatch, useAppSelector } from "../../redux/store";
+import { Libraries } from "@react-google-maps/api/dist/utils/make-load-script-url";
+import { getLocation, setLocation } from "../../redux/slices/locationSlice";
+import REACT_APP_GOOGLE_API_KEY from "../../env";
+import styled from "styled-components";
+
+//TODO: Move out of component because it causes performance issue
+const GOOGLE_LIBRARIES: Libraries = ["places"];
+
+const googleMapStyle: CSSProperties = {
+  width: "80%",
+  height: "10em"
+};
+
+const StyledContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+//TODO: Use built in material ui styles
+const autocompleteStyle: CSSProperties = {
+  boxSizing: "border-box",
+  border: "1px solid transparent",
+  width: "240px",
+  height: "32px",
+  padding: "0 12px",
+  borderRadius: "3px",
+  boxShadow: "0 2px 6px rgba(0, 0, 0, 0.3)",
+  fontSize: "14px",
+  outline: "none",
+  textOverflow: "ellipses",
+  position: "absolute",
+  left: "50%",
+  marginLeft: "-120px",
+};
+
+/**
+ * For autocomplete to populate when the user actually selects something
+ */
+let currLocation: any = null;
+
+const onAutoCompleteLoad = (autocomplete: any) => {
+  console.log("autocomplete: ", autocomplete);
+
+  currLocation = autocomplete;
+};
+
+const Map = () => {
+  const dispatch = useAppDispatch();
+  const location = useAppSelector((state) => state.location);
+
+  useEffect(() => {
+    dispatch(() => getLocation());
+  }, [dispatch]);
+
+  const [ctr, setCtr] = React.useState({
```

Comment:
is there a reason we're keeping track of coordinate state both in this component and the store?

Reply:
yeah the GoogleMap takes an object with "lat" and "lng"...


LNG.....
WHYYYYYYYYYYYYYYYY

Reply:
WE CAN DO THAT TOO HNGGGGGG
or we can just adjust state of location selector
or 
or
nvm its merged